conans/client/rest_client_local_recipe_index.py

```diff
@@ -213,6 +213,14 @@ def get_recipes_references(self, pattern):
                 raise ConanException(f"Corrupted repo, folder {r} without 'config.yml'")
             versions = config_yml["versions"]
             for v in versions:
+                try:
+                    pattern_ref = RecipeReference.loads(pattern)
+                    # We don't care about user/channel, _search_remote_recipes is checking for those at a later point,
+                    # just add all candidates for now
+                    pattern = f"{pattern_ref.name}/{pattern_ref.version}"
+                except:
+                    # pattern = pattern
```

Comment:
I dont think this can actually happen, pattern is already str of a RecipeReference afaik from https://github.com/conan-io/conan/blob/2.10.0/conans/client/graph/range_resolver.py#L55

Reply:
It is fine to exclude the user/channel for this initial check, but then it has to be re-checked again after the actual ``conanfile`` has been loaded and to match the real user/channel defined by recipe?

Reply:
As far as I can tell, this is being done in `_search_remote_recipes` https://github.com/conan-io/conan/blob/develop2/conans/client/graph/range_resolver.py#L80 as it expects that the search has ignored the user/channel

Reply:
There it contains a:
```
# TODO: This is still necessary to filter user/channel, until search_recipes is fixed
```
So basically trying this to return the right thing and not rely on RangeResolved to fix it later?